File: C#2/03.Methods/04.AppearanceCount/AppearanceCount.cs
/*Problem 4. Appearance count

Write a method that counts how many times given number appears in given array.
Write a test program to check if the method is workings correctly.*/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class AppearanceCount
{
    public static void Main()
    {
        int[] numbers = { 1, 2, 3, 4, 5, 6, 7, 1, 2, 3, 4, 5, 6, 7, 1, 2, 3, 4, 5, 6, 7, 1, 2, 3, 4, 5, 6, 7, 1, 2, 3, 4, 5, 6, 7 };
        int number = 4;

        int count = GetAppearanceCount(numbers, number);
        Console.WriteLine(count);
    }

    private static int GetAppearanceCount(int[] numbers, int number)
    {
        int count = 0;

        foreach (var num in numbers)
        {
            if (num == number)
            {
                count++;   
            }
        }

        return count;
    }
}

File: JavaScript Fundamentals/08.Functions/scripts/appearance-count.js
/*<li> Write a function that counts how many times given number appears in given array.</li>
 <li> Write a test function to check if the function is working correctly.</li>*/
function countAppearances(array, key) {
    var num,
        count = 0;

    for (num in array) {
        if (array[num] === key) {
            count += 1;
        }
    }

    return count;
}

function testCountAppearances() {
    if (countAppearances([1, 2, 3, 3, 1, 2, 5, 7, 5, 3], 3) !== 3) {
        return false;
    }
    else if (countAppearances([1, 2, 3, 3, 1, 2, 5, 7, 5, 3], 5) !== 2) {
        return false;
    }
    else if (countAppearances([1, 2, 3, 3, 1, 2, 5, 7, 5, 3], 7) !== 1) {
        return false;
    }
    else if (countAppearances([1, 2, 3, 3, 1, 2, 5, 7, 5, 3], 1) !== 2) {
        return false;
    }
    else if (countAppearances([1, 2, 3, 3, 1, 2, 5, 7, 5, 3], 2) !== 2) {
        return false;
    }
    else {
        return true;
    }
}
var word = testCountAppearances() ? '' : 'not ';
console.log('The function is ' + word + 'working correctly!');